tests/src/lib.rs

```diff
@@ -207,3 +210,23 @@ async fn ws_more_request_than_buffer_should_not_deadlock() {
 		req.await.unwrap();
 	}
 }
+
+#[tokio::test]
+async fn wss_works() {
+	let client = WsClient::new(WsConfig::with_url("wss://kusama-rpc.polkadot.io")).await.unwrap();
+	let response: String = client.request("system_chain", Params::None).await.unwrap();
+	assert_eq!(&response, "Kusama");
+}
+
+#[tokio::test]
+async fn ws_with_non_ascii_url_doesnt_hang_or_panic() {
+	let err = WsClient::new(WsConfig::with_url("wss://♥♥♥♥♥♥∀∂")).await;
+	assert!(matches!(err, Err(Error::TransportError(_))));
+}
+
+#[tokio::test]
+async fn http_with_non_ascii_url_doesnt_hang_or_panic() {
+	let client = HttpClient::new("http://♥♥♥♥♥♥∀∂", HttpConfig::default()).unwrap();
```

Comment:
https://docs.rs/url/2.2.0/url/enum.Host.html#variant.Domain is the reason why it's doesn't fail parsing the URL in the constructor.

Reply:
and in the `WebSocket` case the host is resolved to a `SockAddr` because an actual connection is established in the constructor.

Reply:
TIL!